variables.tf

```diff
@@ -115,7 +115,7 @@ variable "node_selector" {
 variable "aks_connector_enabled" {
   type        = bool
   description = "Controls whether the Ocean AKS Connector should be deployed (requires a valid `acd_identifier`)"
-  default     = true
+  default     = false
```

Comment:
Thank you for your contribution, @t3mi. Would you mind explaining this change?

Reply:
Sure, thank you for review. As null check for acd_identifier was removed, value was switched to false to have the same set of resources deployed for clouds other then Azure. For Azure it should be explicitly defined which is kind of a breaking change.

Reply:
Thank you for responding. What is the motivation for introducing a breaking change that doesn't change the actual behavior? The set of resources will remain the same even when you keep the default value of the `aks_connector_enabled` variable due to the necessity of setting the `acd_identifier` (which is null by default). What am I missing?

Reply:
Currently, I can't set `acd_identifier` to use value from random provider with dependencies for kubernetes cluster changed properties (for example, kubernetes version) so that new information get updated in the spotinst portal. In this PR I've removed the check for null which means for every cloud aks connector job would be deployed even if its not needed. If its ok, Ill revert the change for highlighted value.

Reply:
@liranp should I revert the change for default value?

Reply:
hi @liranp, I have the same issue with acd_identifier variable, because it's evaluated in "count" section isn't possible to use random generated value there.  count evaluate the variable value before the random value is generated.

So, this:
resource "kubernetes_job" "this" {
  count = var.create_controller && var.aks_connector_enabled && var.acd_identifier != null ? 1 : 0
........

Needs to be changed to this:
resource "kubernetes_job" "this" {
  count = var.create_controller && var.aks_connector_enabled ? 1 : 0

Reply:
So, back to the aks_connector_enabled default value to false is required to control when the job is deployed, because today is deployed only when acd_identifier != null (aks_connector_enabled is True by default), and it should be deployed when aks_connector_enabled = True .

Reply:
An example:

```terraform
locals {
  acd_indentifier = "acd-${random_id.this.hex}"
}

resource "random_id" "this" {
  byte_length = 8
}

module "ocean-controller" {
  source = "spotinst/ocean-controller/spotinst"
  # Credentials.
  spotinst_token   = var.spotinst_token
  spotinst_account = var.spotinst_account

  # Configuration.
  cluster_identifier = var.cluster_name
  aks_connector_enabled = true
  acd_identifier     = local.acd_indentifier

}
```

In this example, if I use the module with var.acd_identifier != null evaluation in count inside of kubernetes_job resource, it's fails.

Reply:
Hi @liranp, any chance to take a look at this?

Reply:
@Lironrad Are you able to review?

Reply:
Hi @Lironrad Please your help here